sdk/mobile-center-updates/src/main/java/com/microsoft/azure/mobile/updates/Updates.java

```diff
@@ -701,8 +771,84 @@ public void onCancel(DialogInterface dialog) {
                 completeWorkflow(releaseDetails);
             }
         });
-        mUpdateDialog = dialogBuilder.create();
-        mUpdateDialog.show();
+        mUpdateDialog = showAndRememberDialogActivity(dialogBuilder);
+    }
+
+    /**
+     * Show unknown sources dialog. This can be called multiple times if clicking on HOME and app resumed
+     * (it could be resumed in another activity covering the previous one).
+     */
+    @UiThread
+    private synchronized void showUnknownSourcesDialog() {
+
+        /* Check if we need to replace dialog. */
+        if (!shouldRefreshDialog(mUnknownSourcesDialog)) {
+            return;
+        }
+        MobileCenterLog.debug(LOG_TAG, "Show new unknown sources dialog.");
+
+        /*
+         * We invite user to go to setting and will navigate to setting upon clicking,
+         * but no monitoring is possible and application will be left.
+         * We want consistent behavior whether application is killed in the mean time or not.
+         * Not changing any state here provide that consistency as a new update dialog
+         * will be shown when coming back to application.
+         *
+         * Also for buttons and texts we try do to the same as the system dialog on standard devices.
+         */
+        AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(mForegroundActivity);
+        dialogBuilder.setMessage(R.string.mobile_center_updates_unknown_sources_dialog_message);
+        final ReleaseDetails releaseDetails = mReleaseDetails;
+        dialogBuilder.setNegativeButton(android.R.string.cancel, new DialogInterface.OnClickListener() {
```

Comment:
Used method instead of field, its a closure using a snapshot of `releaseDetails` so we need the `new` or it would need more complex refactoring with a completeWorflow accepting a dialog as parameter for state check.